vespalib/src/vespa/vespalib/data/slime/slime.h

```diff
@@ -53,75 +51,50 @@ class Slime
     typedef slime::Cursor        Cursor;
     typedef slime::Inspector     Inspector;
 
-    SymbolTable::UP              _names;
-    Stash::UP                    _stash;
-    RootValue                    _root;
+    std::unique_ptr<SymbolTable>   _names;
+    std::unique_ptr<Stash>         _stash;
+    RootValue                      _root;
 
 public:
     typedef std::unique_ptr<Slime> UP;
     class Params {
     private:
-        SymbolTable::UP  _symbols;
-        size_t           _chunkSize;
+        std::unique_ptr<SymbolTable>  _symbols;
+        size_t                        _chunkSize;
     public:
-        Params() : Params(std::make_unique<SymbolTable>()) { }
-        explicit Params(SymbolTable::UP symbols) : _symbols(std::move(symbols)), _chunkSize(4096) { }
+        Params();
+        explicit Params(std::unique_ptr<SymbolTable> symbols);
+        Params(Params &&) noexcept;
+        ~Params();
         Params & setChunkSize(size_t chunkSize) {
             _chunkSize = chunkSize;
             return *this;
         }
         size_t getChunkSize() const { return _chunkSize; }
-        SymbolTable::UP detachSymbols() { return std::move(_symbols); }
+        std::unique_ptr<SymbolTable> detachSymbols();
     };
     /**
      * Construct an initially empty Slime object.
      **/
-    explicit Slime(Params params = Params()) :
-        _names(params.detachSymbols()),
-        _stash(std::make_unique<Stash>(params.getChunkSize())),
-        _root(_stash.get())
-    { }
+    explicit Slime(Params params = Params());
 
     ~Slime();
 
-    Slime(Slime &&rhs) noexcept :
-        _names(std::move(rhs._names)),
-        _stash(std::move(rhs._stash)),
-        _root(std::move(rhs._root))
-    {
-    }
+    Slime(Slime &&rhs) noexcept;
+    Slime &operator=(Slime &&rhs);
```

Comment:
Consider noexcept specifier for move assignment.  Similar for RootValue move constructor and move assignment.

Reply:
Good point, thanks